File: app/src/main/java/com/cogu/spylook/model/decorators/SpacingItemDecoration.java
package com.cogu.spylook.model.decorators;

import android.content.Context;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.cogu.spylook.R;

public class SpacingItemDecoration extends RecyclerView.ItemDecoration {
    private Context context;
    public SpacingItemDecoration(Context context) {
        this.context = context;
    }

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        int vertical = context.getResources().getDimensionPixelSize(R.dimen.spacing_vertical);
        int horizontal = context.getResources().getDimensionPixelSize(R.dimen.spacing_horizontal);
        outRect.set(horizontal, vertical, horizontal, vertical);
    }
}

File: app/src/main/java/com/cogu/spylook/model/utils/decorators/SpacingItemDecoration.kt
package com.cogu.spylook.model.utils.decorators

import android.content.Context
import android.graphics.Rect
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ItemDecoration
import com.cogu.spylook.R

class SpacingItemDecoration(private val context: Context) : ItemDecoration() {
    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        val vertical = context.getResources().getDimensionPixelSize(R.dimen.spacing_vertical)
        val horizontal = context.getResources().getDimensionPixelSize(R.dimen.spacing_horizontal)
        outRect.set(horizontal, vertical, horizontal, vertical)
    }
}
